pulsar4s-core/src/main/scala/com/sksamuel/pulsar4s/FutureAsyncHandler.scala

```diff
@@ -92,4 +99,16 @@ class FutureAsyncHandler(implicit ec: ExecutionContext) extends AsyncHandler[Fut
 
   override def send[T](builder: TypedMessageBuilder[T]): Future[MessageId] =
     builder.sendAsync().toScala.map(MessageId.fromJava)
+
+  override def withTransaction[T](create: () => CompletableFuture[Transaction], action: Transaction => Future[T]): Future[T] = {
+    create().toScala.flatMap { txn =>
+      action(txn).transformWith {
+        case Success(value) =>
+          txn.commit().toScala.transform(_ => Success(value))
```

Comment:
I guess the question here is whether the app should be able to roll back a Pulsar operation because of a failure in another system, but that looks like something that I guess you would account for by failing the `action` right?  So in other words, if we wanted to do something in Pulsar, then with the result do something in a database, as long as the database failure resulted in failing the `action` return value we'd be able to roll back the Pulsar part of the transaction...

Reply:
So I added another API that I think gives you more flexibility. Now you can pass a `TransactionContext => F[Either[E, A]]` and we'll abort the transaction if we get a `Left`. This keeps the more functional API while also giving the user a way to explicitly abort the transaction without having to return a failed `Future`/`Task`.